solarforecastarbiter/reference_forecasts/tests/test_fx_utils.py

```diff
@@ -256,40 +281,104 @@ def test_get_data_start_end_labels_obs_avg_fx_instant(site_metadata):
         run_length=pd.Timedelta('1d'),
         interval_label='instant')
     with pytest.raises(ValueError) as excinfo:
-        utils.get_data_start_end(observation, forecast, run_time)
+        utils.get_data_start_end(observation, forecast, run_time, issue_time)
     assert 'made from interval average obs' in str(excinfo.value)
 
 
-@pytest.mark.parametrize("variable,expected_start,expected_end", [
+@pytest.mark.parametrize('rl,expected_start,expected_end,rt,lt', [
+    ('1d', '20190409T0000Z', '20190410T0000Z', '20190410T2230Z', '1h'),
+    ('1d', '20190409T0000Z', '20190410T0000Z', '20190410T0001Z', '1h'),
+    ('1d', '20190409T0000Z', '20190410T0000Z', '20190410T0001Z', '25h'),
+    ('1d', '20190409T0000Z', '20190410T0000Z', '20190410T0001Z', '49h'),
+    ('1d', '20190408T2300Z', '20190409T2300Z', '20190410T2230Z', '24h'),
+    # quirky that 23h ahead uses more recent data than 1h ahead, but
+    # timestamps are maintained and end is before issue time
+    ('1d', '20190409T2200Z', '20190410T2200Z', '20190410T2230Z', '23h'),
+    ('2d', '20190408T0000Z', '20190410T0000Z', '20190410T2230Z', '1h'),
+    ('36h', '20190409T0000Z', '20190410T1200Z', '20190410T2300Z', '1h'),
+    # run time is too early compared to init time so data_end > run_time
+    # and that's ok.
```

Comment:
I don't understand why that's ok. There would not be enough data to issue a complete forecast.

Reply:
It's ok because run persistence raises that value error. Day/week ahead persistence data is set based on the forecast parameters, so this is basically just testing that you tried to make the forecast too early